plugins/woocommerce/includes/gateways/bacs/class-wc-gateway-bacs.php

```diff
@@ -64,6 +64,7 @@ public function __construct() {
 		);
 
 		// Actions.
+		add_action( 'woocommerce_update_options_payment_gateways_' . $this->id, array( $this, 'add_admin_options_tracking' ) );
```

Comment:
This will require us to add this action in every gateway which is prone to error and won't track third party gateways.

Since the ajax request calls `update_option()` ( https://github.com/woocommerce/woocommerce/blob/f3927c786a62b8bd5ca7ba7c4ef05f2ebdf0b787/plugins/woocommerce/includes/class-wc-ajax.php#L3095-L3099 ), maybe we could just plug in to the option update hook and check if the value has changed.

There's even some precedent for this and we may be able to add this as an allowed settings in this class - https://github.com/woocommerce/woocommerce/blob/f3927c786a62b8bd5ca7ba7c4ef05f2ebdf0b787/plugins/woocommerce/includes/tracks/events/class-wc-settings-tracking.php#L57-L82

Reply:
Nice catch @joshuatf! I added commit ca88734 to fix the issue you mentioned here. The solution I found was a little different from what you suggested but I think it will work well.